File: src/main/java/com/raychenon/leetcode/besttimetobuyandsellstock/BestTimetoBuyandSellStock2.java
package com.raychenon.leetcode.besttimetobuyandsellstock;

/**
 * User: raychenon
 * Date: 5/4/2020
 * https://leetcode.com/problems/best-time-to-buy-and-sell-stock-ii/
 */
public class BestTimetoBuyandSellStock2 {

    public int maxProfit(int[] prices) {
        int profit = 0;
        for (int i = 0; i < prices.length-1; i++) {
            if (prices[i + 1] - prices[i] > 0) {
                profit += prices[i + 1] - prices[i];
            }
        }
        return profit;
    }
}

File: python3/algorithms/leetcode/buyandsell/best_time_to_buy_and_sell_stock2.py
#!/bin/python3
from typing import List


def max_profit(prices: List[int]) -> int:
    """
    https://leetcode.com/problems/best-time-to-buy-and-sell-stock-ii/
    """
    profit = 0
    for i in range(0, len(prices) - 1):
        if prices[i + 1] > prices[i]:
            profit += prices[i + 1] - prices[i]

    return profit


def test_max_profit():
    assert max_profit([7, 1, 5, 3, 6, 4]) == 7
    assert max_profit([1, 2, 3, 4, 5]) == 4
    assert max_profit([7, 6, 4, 3, 1]) == 0
    assert max_profit([7, 6, 4, 3, 1, 2]) == 1
    assert max_profit([]) == 0
